packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -312,23 +312,25 @@ export function createHooksInternal<
               const [queryKey, ...rest] = args;
 
               return queryClient.refetchQueries(
-                getArrayQueryKey(queryKey),
+                getArrayQueryKey(queryKey, 'any'),
```

Comment:
I think this is prefect and that we don't need to allow granular invalidation for now, I don't see why anyone would _not_ want to refetch the same query across both methods of fetching it.